### PR TITLE
machine1: Implement CloneImage

### PR DESCRIPTION
**Pull request by Xuanwo**
This commit added CloneImage support in dbus, user can clone their
images via dbus API.

### CODE REVIEW

File: machine1/dbus_test.go [hunk 1]
```diff
@@ -157,7 +157,10 @@ func generateRandomLabel(n int) string {
 
 func TestImages(t *testing.T) {
 	imageName := machinePrefix + generateRandomLabel(8)
-	imagePath := filepath.Join("/var/lib/machines", imageName)
+	imagePath := filepath.Join("/var/lib/machines", imageName+".raw")
+
+	newImageName := machinePrefix + generateRandomLabel(8)
+	newImagePath := filepath.Join("/var/lib/machines", newImageName+".raw")
```
Comment (lucab) [inline machine1/dbus_test.go]: Is this used anywhere?

File: machine1/dbus_test.go [hunk 2]
```diff
@@ -181,4 +184,14 @@ func TestImages(t *testing.T) {
 	if len(listImages) < 1 {
 		t.Fatalf("did not find any image")
 	}
+
+	cloneErr := conn.CloneImage(imageName, newImageName, false)
+	if cloneErr != nil {
+		t.Fatal(cloneErr)
+	}
+	defer os.Remove(newImagePath)
```
Comment (lucab) [inline machine1/dbus_test.go]: I think you may want to `RemoveImage` here, not removing by FS path.

**Comment by lucab** [comment]
@Xuanwo thanks for the PR! I've left a few comments regarding the testing logic.

File: machine1/dbus_test.go (hunk 1 above)
Comment (Xuanwo) [reply machine1/dbus_test.go]: It's used in CloneImage test.

File: machine1/dbus_test.go (hunk 2 above)
Comment (Xuanwo) [reply machine1/dbus_test.go]: The logic is borrowed from old tests, maybe we should refactor them all?

File: machine1/dbus_test.go (hunk 1 above)
Comment (lucab) [reply machine1/dbus_test.go]: I'm sorry, I'm not following. `newImagePath` is used exactly once in a `os.Remove`; who is creating and populating that file?

Comment (Xuanwo) [reply machine1/dbus_test.go]: conn.CloneImage will copy the image to new place.

File: machine1/dbus_test.go (hunk 2 above)
Comment (lucab) [reply machine1/dbus_test.go]: The old test is creating the file directly, while here it happens implicitly (by machined). That wasn't clear to me first at first.

If you have time, I'd be happy to see this test doing the following before exiting:
 * doing `defer os.Remove()` on `imagePath` and `newImagePath`
 * calling `RemoveImage` on both `imageName` and `newImageName`
 * checking that  `imagePath` and `newImagePath` do not exist anymore

Comment (Xuanwo) [reply machine1/dbus_test.go]: OK, I will refactor the logic for test.

**Pull request closed without merging**